src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/TranscodingReadStream.netcoreapp.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace System.Net.Http.Json
+{
+    internal sealed partial class TranscodingReadStream : Stream
+    {
+        private Memory<byte> _byteBuffer;
+        private Memory<char> _charBuffer;
+        private Memory<byte> _overflowBuffer;
+
+        internal int ByteBufferCount => _byteBuffer.Length;
+        internal int CharBufferCount => _charBuffer.Length;
+        internal int OverflowCount => _overflowBuffer.Length;
+
+        private void InitializeBuffers() { }
+
+        public override Task<int> ReadAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+        {
+            if (buffer == null)
+            {
+                throw new ArgumentNullException(nameof(buffer));
+            }
+
+            if (offset < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(offset));
+            }
+
+            if (count < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(count));
+            }
+
+            if (buffer.Length - offset < count)
+            {
+                throw new ArgumentException(SR.Argument_InvalidOffLen);
+            }
```

Comment:
The `Memory` ctor performs these checks; shouldn't need them here unless parameter names are different.